Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeInfoWrapper.cs

```diff
@@ -0,0 +1,562 @@
+        {
+            int hr = _target_ITypeInfo.GetFuncDesc(index, ppFuncDesc);
+            if (ComHelper.HRESULT_FAILED(hr)) return HandleBadHRESULT(hr);
+
+            if (_target_ITypeInfoAlternate != null)
+            {
+                var pFuncDesc = StructHelper.ReadStructureUnsafe<IntPtr>(ppFuncDesc);
+                var funcDesc = StructHelper.ReadStructureUnsafe<ComTypes.FUNCDESC>(pFuncDesc);
+
+                // Populate wFuncFlags from the alternative typeinfo provided by VBA
+                // The alternative typeinfo is not as useful as the main typeinfo for most things, but does expose wFuncFlags
+                // The list of functions appears to be in the same order as the main typeinfo.  
+                using (var funcDescAlternatePtr = AddressableVariables.CreatePtrTo<ComTypes.FUNCDESC>())
+                {
+                    var hr2 = _target_ITypeInfoAlternate.GetFuncDesc(index, funcDescAlternatePtr.Address);
+                    if (!ComHelper.HRESULT_FAILED(hr2))
+                    {
+                        var funcDescAlternate = funcDescAlternatePtr.Value.Value;    // dereference the ptr, then the content
+
+                        //sanity check
+                        if (funcDescAlternate.memid == funcDesc.memid)
+                        {
+                            funcDesc.wFuncFlags = funcDescAlternate.wFuncFlags;
+                        }
+                        else
+                        {
+                            // FIXME log
+                        }
+                        _target_ITypeInfoAlternate.ReleaseFuncDesc(funcDescAlternatePtr.Value.Address);
+
+                        Marshal.StructureToPtr(funcDesc, pFuncDesc, false);
+                    }
+                }
+            }
+
+            return hr;
+        }
+
+        public override int GetVarDesc(int index, IntPtr ppVarDesc)
+        {
+            int hr = _target_ITypeInfo.GetVarDesc(index, ppVarDesc);
+            if (ComHelper.HRESULT_FAILED(hr)) return HandleBadHRESULT(hr);
+
+            var pVarDesc = StructHelper.ReadStructureUnsafe<IntPtr>(ppVarDesc);
+            var varDesc = StructHelper.ReadStructureUnsafe<ComTypes.VARDESC>(pVarDesc);
+            if (varDesc.memid == (int)KnownDispatchMemberIDs.MEMBERID_NIL)
+            {
+                // constants are not reported correctly in VBA type infos.  They all have MEMBERID_NIL set.
+                // we will provide fake DispIds and names to satisfy parsers.  Shit but works for now.
+                varDesc.memid = (int)(_ourConstantsDispatchMemberIDRangeStart + index);
+                Marshal.StructureToPtr(varDesc, pVarDesc, false);
+            }
+            else
+            {
+                // Unlike GetFuncDesc, we can't get the wVarFlags for fields from the alternative VBA ITypeInfo
+                // because GetVarDesc() hard crashes on the alternative typeinfo
+                /*
```

Comment:
The block comment & non-working code should be deleted?

Reply:
yup, will do.